Automattic/jetpack
# Display the Correct Newsletter Gated text when Stripe is Disconnected

Relates to  https://github.com/Automattic/jetpack/issues/29878

## Proposed changes:

Made newsletter display the correct text when Stripe is disconnected. Previously we displayed: "This post is for paid subscribers" on a newsletter marked as "paid subscribers" when Stripe was disconnected from the site. Now we check that Stripe is connected and the newsletter is marked as "paid subscribers" before displaying the "This post is for paid subscribers" text.

| Stripe Disconnected  | Stripe Connected |
| ------------- | ------------- |
| <img width="910" alt="Screenshot 2023-04-03 at 12 51 13 PM" src="https://user-images.githubusercontent.com/7076981/229581868-4b687676-3e6d-4325-83e5-adfa34aba0a1.png">  | <img width="856" alt="Screenshot 2023-04-03 at 12 51 46 PM" src="https://user-images.githubusercontent.com/7076981/229581871-9a8fa939-dd2a-4cfb-9f3c-ff1f8c66cd88.png"> |

### Other information:



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Create a post for "paid subscribers" with Stripe connected
* Verify that the locked content in an incognito window displays: "This post is for paid subscribers"
* Disconnect Stripe
* Verify the locked content in an incognito displays displays: "This post is for subscribers"
